benchmark/opperf/opperf.py

```diff
@@ -0,0 +1,133 @@
+    mxnet_operator_benchmark_results.append(run_mx_binary_element_wise_operators_benchmarks(ctx=ctx,
+                                                                                            dtype=dtype))
+
+    # Run all GEMM operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_gemm_operators_benchmarks(ctx=ctx,
+                                                                          dtype=dtype))
+
+    # Run all Random sampling operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_mx_random_sampling_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # Run all Reduction operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_mx_reduction_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # ************************ MXNET NN OPERATOR BENCHMARKS ****************************
+
+    # Run all basic NN operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_nn_basic_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # Run all Activation operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_activation_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # Run all Pooling operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_pooling_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # Run all Convolution operations benchmarks with default input values
+    mxnet_operator_benchmark_results.append(run_convolution_operators_benchmarks(ctx=ctx, dtype=dtype))
+
+    # ****************************** PREPARE FINAL RESULTS ********************************
+    final_benchmark_result_map = merge_map_list(mxnet_operator_benchmark_results)
+    return final_benchmark_result_map
+
+
+def _parse_mxnet_context(ctx):
+    if not ctx:
+        raise ValueError("Context cannot be null or empty")
+
+    if ctx.lower() in ['cpu', 'gpu']:
+        return mx.context.Context(ctx)
+    elif ctx.lower().startwith('gpu('):
+        device_id = int(ctx[4:-1])
+        return mx.gpu(device_id)
+
+
+if __name__ == '__main__':
+    # CLI Parser
+
+    # 1. GET USER INPUTS
+    parser = argparse.ArgumentParser(
+        description='Run all the MXNet operators (NDArray) benchmarks')
+
+    parser.add_argument('--ctx', type=str, default='cpu',
+                        help='Global context to run all benchmarks. By default, cpu on a '
+                             'CPU machine, gpu(0) on a GPU machine. '
+                             'Valid Inputs - cpu, gpu, gpu(0), gpu(1)...')
+    parser.add_argument('--dtype', type=str, default='float32', help='DType (Precision) to run benchmarks. By default, '
+                                                                     'float32. Valid Inputs - float32, float64.')
+    parser.add_argument('--output-format', type=str, default='json',
```

Comment:
Do we need to provide markdown format? Can we just let user use https://www.npmjs.com/package/json2md if they need md format?

Reply:
Yes. I thought, usually we want to run this utility and generate a kind of report easy to view / consume manually hence added.